test/functorch/test_vmap.py

```diff
@@ -4793,6 +4794,20 @@ def f(x, gy):
 
         self.vmap_outplace_test(f, (x, gy), {}, in_dims=(None, 0))
 
+    @onlyCUDA
+    @parametrize("inplace", [True, False])
+    def test_0d_tensor_index_put(self, device, inplace):
+        def f(t, idx, v):
+            fn = torch.index_put_ if inplace else torch.index_put
+            return fn(t, idx, v)
+
+        t = torch.zeros((1, 5), device="cuda")
+        idx = torch.tensor([1, 3])
+        v = torch.tensor(1, dtype=t.dtype, device="cpu")
+
+        expected = torch.tensor([[0, 1, 0, 1, 0]], dtype=t.dtype)
+        self.assertEqual(expected, vmap(f, in_dims=(0, None, None))(t, (idx,), v))
```

Comment:
Can you also test the "batched" v case? 
```
t = torch.zeros((N, 5), device="cuda")
idx = torch.tensor([1, 3])
v = torch.tensor([3.15, 2.71])

# if t[0].index_put_(idx, v[0]) works, then vmap(Tensor.index_put_, (0, None, 0))(t, idx, v) should also work.
```

Reply:
`v` stops being a 0d tensor if we batch it.

Reply:
```python
  File "/home/guilhermeleobas/git/pytorch/test/functorch/test_vmap.py", line 4814, in test_0d_tensor_index_put
    self.assertEqual(expected, vmap(f, in_dims=(0, None, 0))(t, (idx,), v))
  File "/home/guilhermeleobas/git/pytorch/torch/_functorch/apis.py", line 201, in wrapped
    return vmap_impl(
  File "/home/guilhermeleobas/git/pytorch/torch/_functorch/vmap.py", line 331, in vmap_impl
    return _flat_vmap(
  File "/home/guilhermeleobas/git/pytorch/torch/_functorch/vmap.py", line 48, in fn
    return f(*args, **kwargs)
  File "/home/guilhermeleobas/git/pytorch/torch/_functorch/vmap.py", line 480, in _flat_vmap
    batched_outputs = func(*batched_inputs, **kwargs)
  File "/home/guilhermeleobas/git/pytorch/test/functorch/test_vmap.py", line 4802, in f
    return fn(t, idx, v)
RuntimeError: batched == nullptr INTERNAL ASSERT FAILED at "/home/guilhermeleobas/git/pytorch/aten/src/ATen/functorch/Interpreter.cpp":99, please report a bug to PyTorch.
```

Reply:
```py
t = torch.zeros((2, 5), device="cuda")
idx = torch.tensor([1, 3])
v = torch.tensor([3.15, 2.71])
vmap(Tensor.index_put_, (0, None, 0))(t, [idx], v)
```
The above should pass. The argument is that because `t[0].index_put_([idx], v[0])` (which it does), then the above should pass.

Now, v does stop being a 0D tensor, but from the perspective of vmap **it is** a 0D tensor.

Reply:
(you can merge this first and fix this in a follow-up)